model/labels.go

```diff
@@ -18,11 +18,104 @@
 package model
 
 import (
+	"strconv"
 	"strings"
 
 	"github.com/elastic/beats/v7/libbeat/common"
 )
 
+type Labels map[string]LabelValue
+
+type LabelValue struct {
+	Value  string
+	Values []string
+}
+
+func (l Labels) MaybeSet(k string, v interface{}) bool {
```

Comment:
Looks like Labels.MaybeSet and NumericLabels.MaybeSet are only used by modeldecoderutil, and one place in processor/otel which could use Set + strconv. Can we please move the function there? I'd like to limit our use of reflection as much as possible.